src/vorta/views/main_window.py

```diff
@@ -106,10 +108,46 @@ def set_progress(self, text=''):
         self.progressText.setText(text)
         self.progressText.repaint()
 
-    def set_log(self, text=''):
+    def set_log(self, text='', ret={}):
         self.logText.setText(text)
         self.logText.repaint()
 
+        msgid = ret.get('msgid')
+        repo_url = ret.get('repo_url')
+        cmd = ret.get('cmd')
+        if cmd is not None and cmd != 'init':
```

Comment:
What you did here is important because it allows us to react to all kinds of Borg events in the future. The function will also grow big, so I suggest to move it out of `main_window.py`.